cluster-autoscaler/cloudprovider/gce/cache.go

```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"sync"
+
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+
+	"github.com/golang/glog"
+	gce "google.golang.org/api/compute/v1"
+)
+
+// GceCache is used for caching cluster resources state.
+type GceCache struct {
+	// Cache content.
```

Comment:
I know it wasn't there before, but I'd like some general comment about what we're caching, why we need it and how all this stuff works. Hopefully it would make it easier for people to approach cloudprovider code in future.

Reply:
Done